### PR TITLE
#103. Added validation rules when adding/editing an availability slot

### PR DESCRIPTION
**Pull request by Ymkavi**
Screenshot 2022-08-19 161701

### CODE REVIEW

File: frontend/src/app/modules/availability/new-slot/general/general.component.ts [hunk 1]
```diff
@@ -81,6 +82,12 @@ export class GeneralComponent implements OnInit {
 
     public addAdvanced: boolean = false;
 
+    public meetingNameInputPattern = "^[A-Za-z0-9-']*$";
+
+    public naturalNumberInputPattern = '^[1-9][0-9]*$';
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/general/general.component.ts]: Could you please move this to our shared file with regexes?)

File: frontend/src/app/modules/availability/new-slot/new-availability/new-availability.component.ts [hunk 2]
```diff
@@ -58,4 +58,12 @@ export class NewAvailabilityComponent implements OnInit {
             this.slot.isEnabled = !this.slot.isEnabled;
         }
     }
+
+    public saveChanges() {
+        if (this.index === 0 && this.generalComponent.generalForm.invalid) {
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/new-slot/new-availability/new-availability.component.ts]: Why do you check the index here?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/new-slot/new-availability/new-availability.component.ts (hunk 2 above)
Comment (Ymkavi) [reply frontend/src/app/modules/availability/new-slot/new-availability/new-availability.component.ts]: To allow to "save changes" on other tabs even if there is invalid data on the general tab left

Comment (tatianahutii) [reply frontend/src/app/modules/availability/new-slot/new-availability/new-availability.component.ts]: Do we need this? we save all data in one request so data in all tabs should be valid

Comment (Ymkavi) [reply frontend/src/app/modules/availability/new-slot/new-availability/new-availability.component.ts]: Ok, I agree

**Review by tatianahutii — COMMENTED** [review]
LGTM

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**